packages/backend-common/src/reading/GitlabUrlReader.ts

```diff
@@ -256,4 +256,69 @@ export class GitlabUrlReader implements UrlReader {
     const { host, token } = this.integration.config;
     return `gitlab{host=${host},authed=${Boolean(token)}}`;
   }
+
+  async getGitlabFetchUrl(target: string): Promise<string> {
+    // If the target is a raw API url then trust that no parsing is needed
+    if (target.includes('/api/v4/')) {
+      return target;
+    }
+    // If the target is for a job artifact then go down that path
+    if (target.includes('/-/jobs/artifacts/')) {
```

Comment:
It's always a bit risky to do this kind of raw string comparison without any knowledge of _where_ in the URL path it appears, but ... I guess the risk here is fairly low anyway, that it'd happen to appear farther down the URL as part of the file path or something.

Reply:
I'll think more on this. I can at least accept a `url` and look at the `pathname` instead of a raw string.